tests/phpunit/tests/rest-api/wpRestTemplateRevisionsController.php

```diff
@@ -558,4 +558,16 @@ public function test_delete_item_not_found() {
 		$response = rest_get_server()->dispatch( $request );
 		$this->assertErrorResponse( 'rest_post_invalid_parent', $response, WP_Http::NOT_FOUND );
 	}
+
+	/**
+	 * @covers WP_REST_Template_Revisions_Controller::get_items_permissions_check
+	 * @ticket 61113
+	 */
+	public function test_get_item_permissions_check_without_parent() {
+		$request    = new WP_REST_Request( 'GET' );
+		$controller = new WP_REST_Template_Revisions_Controller( self::PARENT_POST_TYPE );
+		$result     = $controller->get_item_permissions_check( $request );
```

Comment:
Instead of instantiating `WP_REST_Template_Revisions_Controller` directly, e usually assemble a valid `WP_REST_Request` and then use `rest_get_server()->dispatch( $request );`. See the tests above for precedence.

That said, how do you get a missing `parent` if you make a regular request like this? 🤔  I can't seem to reproduce the issue.

The bug fix itself looks ok to me, but I think we could make the test more robust.

Reply:
That is exactly the reason I am instantiating the `WP_REST_Template_Revisions_Controller`, otherwise a missing `parent` is hard to get. I found this bug when I was looping through all registered REST routes and calling their `permission_callback` to see if they are publically available (I didn't want to do actual REST requests).

So although you cannot reproduce the error using a REST request, since you can call the `get_item_permissions_check` after instantiating `WP_REST_Template_Revisions_Controller`, I think this bug should be fixed. And I think this is the only way to test it correctly?

Reply:
As I said, the bug fix itself looks ok to me in theory. The different test just seemed unusual, but yeah seems like the only way.

Cc @TimothyBJacobs for 2nd opinion